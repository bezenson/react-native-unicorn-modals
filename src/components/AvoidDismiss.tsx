import type { PropsWithChildren } from 'react';
import React from 'react';
import type { TouchableOpacityProps } from 'react-native';
import { TouchableOpacity } from 'react-native';

interface AvoidDismissProps {
  style?: TouchableOpacityProps['style'];
}

const AvoidDismiss: React.FC<PropsWithChildren<AvoidDismissProps>> = ({ children, style }) => {
  return (
    <TouchableOpacity activeOpacity={1} style={style}>
      {children}
    </TouchableOpacity>
  );
};

export default AvoidDismiss;
