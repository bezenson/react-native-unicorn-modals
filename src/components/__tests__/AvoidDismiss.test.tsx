import React from 'react';
import { render, screen } from '@testing-library/react-native';

import AvoidDismiss from '../AvoidDismiss';

describe('AvoidDismiss', () => {
  it('should match snapshot', () => {
    render(<AvoidDismiss style={{ backgroundColor: 'red' }} />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
});
