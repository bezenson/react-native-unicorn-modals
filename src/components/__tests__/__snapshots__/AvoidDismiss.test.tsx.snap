// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AvoidDismiss should match snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "red",
      "opacity": 1,
    }
  }
/>
`;
